java/arcs/core/analysis/ClaimDeduction.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.analysis
+
+import arcs.core.data.expression.Expression
+
+internal typealias Path = List<String>
+internal typealias Paths = List<Path>
+internal typealias ClaimDerivations = Map<Path, Set<Path>>
```

Comment:
If you are going to keep `Strings`, you might as well keep `AccessPath.Selector` instances.

Reply:
In this latest revision (2020-09-23), I've decided to go with typealiases to Strings once more. 

I am doing this since an access path will be derived from the combination of keys in the nested structure (not yet implemented).